finality-aleph/src/network/build/mod.rs

```diff
@@ -0,0 +1,120 @@
+use std::sync::{atomic::AtomicBool, Arc};
+
+use log::error;
+use sc_client_api::Backend;
+use sc_network::{
+    config::{NetworkConfiguration, ProtocolId},
+    error::Error as NetworkError,
+};
+use sc_network_sync::SyncingService;
+use sc_network_transactions::TransactionsHandlerController;
+use sc_rpc::system::Request as RpcRequest;
+use sc_service::SpawnTaskHandle;
+use sc_transaction_pool_api::TransactionPool;
+use sc_utils::mpsc::TracingUnboundedSender;
+use sp_runtime::traits::{Block, Header};
+use substrate_prometheus_endpoint::Registry;
+
+use crate::{
+    network::{
+        base_protocol::{setup as setup_base_protocol, Service as BaseProtocolService},
+        LOG_TARGET,
+    },
+    BlockHash, BlockNumber, ClientForAleph, ProtocolNetwork,
+};
+
+mod base;
+mod own_protocols;
+mod rpc;
+mod transactions;
+
+// TODO(A0-3576): should no longer be needed outside of here
+use base::network as base_network;
+pub use own_protocols::Networks;
+use rpc::spawn_rpc_service;
+use transactions::spawn_transaction_handler;
+
+/// Components created when spawning the network.
+pub struct NetworkOutput<TP: TransactionPool + 'static> {
+    pub authentication_network: ProtocolNetwork,
+    pub block_sync_network: ProtocolNetwork,
+    // names chosen for compatibility with SpawnTaskParams, get better ones if we ever stop using that
+    pub sync_service: Arc<SyncingService<TP::Block>>,
+    pub tx_handler_controller: TransactionsHandlerController<TP::Hash>,
+    pub system_rpc_tx: TracingUnboundedSender<RpcRequest<TP::Block>>,
+}
+
+/// Start everything necessary to run the inter-node network and return the interfaces for it.
+// TODO(A0-3576): This code should be used.
```

Comment:
Pls add to the comment what are the things startet up here

Reply:
Done.